src/BCFSerialize.cc

```diff
@@ -98,14 +106,38 @@ int bcf_raw_read_from_mem(const char *addr, bcf1_t *v) {
     // bcf_subset, prior to and including bd6ed8b4
     if ( (!v->indiv.l || !v->n_sample) && v->n_fmt ) v->n_fmt = 0;
 
-    memcpy(v->shared.s, &addr[loc], v->shared.l);
+    memcpy(v->shared.s, &buf[start + loc], v->shared.l);
     loc += v->shared.l;
-    memcpy(v->indiv.s, &addr[loc], v->indiv.l);
+    memcpy(v->indiv.s, &buf[start + loc], v->indiv.l);
     loc += v->indiv.l;
 
     return loc;
 }
 
+// Calculate the total length of a record, without unpacking it.
+uint32_t bcf_raw_calc_rec_len(const char *buf, int start, size_t len) {
+    range_check(start + 32, len);
+    uint32_t *x = (uint32_t*) &buf[start];
+
+    assert(x[0] > 0);
+    uint32_t rc = 32 + (x[0] -24) + x[1];
+    return rc;
+}
+
+// Check if the record that starts at memory address [addr] overlaps
+// the range [rng]. The trick is to do this without unpacking the record.
+bool bcf_raw_overlap(const char *buf, int start, size_t len, const range &rng) {
+    range_check(start + 32, len);
+    uint32_t *x = (uint32_t*) &buf[start];
+
+    int32_t rid = x[2];
+    int32_t beg = x[3];
+    int32_t rlen = x[4];
+    int32_t end = beg + rlen;
+
+    return rid == rng.rid && end > rng.beg && beg < rng.end;
+}
```

Comment:
Similarly, it seems like `bcf_raw_calc_rec_len` and `bcf_raw_overlap`, and consequently functions that call them, should in some situations produce Status::Invalid